### PR TITLE
GH actions + testthat 3e

### PR DESCRIPTION
**Pull request by jonkeane**
I've moved over CI to GH actions following (loosely) the r-lib/actions examples. I've left travis and appveyor for now, but those both could be dropped here too. 

I could never figure out what's going on with the [one test that fails on windows on actions but not appveyor](https://github.com/nealrichardson/httptest/blob/master/tests/testthat/test-capture-requests.R#L73). It looks like `m2` — which is based on a mock [recorded above](https://github.com/nealrichardson/httptest/blob/master/tests/testthat/test-capture-requests.R#L12) — has `\r\n` instead of simply `\n` when it is played back. I tried adding in the git setting for crlf (it's still there at the top of the action) and I even tried `gsub`ing out `\r`, etc. to try and get the two to match to no avail (I left the series of commits trying [on a branch](https://github.com/jonkeane/httptest/tree/gh-actions)). I suspect there is something else funky about whitespace, and possibly encoding going on there. But for now, I've punted and only run that test on not-windows.

The 3e tests are running (both for the 2nd and 3rd edition). The 3rd edition doesn't (yet) pass, though I'm working on a branch where it does (for the most part, lots of warnings that need to be cleaned up, and making sure it can pass both 2e and 3e at the same time though that's not strictly necessary since we can choose to only test httptest with either 2e or 3e, I found with dittodb it wasn't too hard to test with both and then I have a bit more confidence I don't accidentally break something when someone is running it under 2e.) and will send those along separately.

### CODE REVIEW

**Comment by nealrichardson** [comment]
Thanks a lot! I'll merge this and we can take up the 3e fixes and the CRLF thing separately (I have some ideas but it will be easier to test them on top of this). 

After I merge, I suggest rebasing your other branches on upstream/master.

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      - name: Show testthat output
+        if: always()
+        run: find check -name 'testthat.Rout*' -exec cat '{}' \; || true
+        shell: bash
+
+      - name: Test coverage
+        if: contains(matrix.config.r, 'release') && runner.os == 'macOS'
+        run: |
+          Rscript -e 'remotes::install_cran("covr")'
+          Rscript -e 'covr::codecov(token = "${{secrets.CODECOV_TOKEN}}")'
```
Comment (nealrichardson) [inline .github/workflows/R-CMD-check.yaml:95]: @jonkeane Do I need to set this token?

Comment (jonkeane) [reply .github/workflows/R-CMD-check.yaml:95]: In my experience it magically works without setting it proactively, but the test will be when travis gets turned off.

Comment (nealrichardson) [reply .github/workflows/R-CMD-check.yaml:95]: Yeah it seems to work. (Turning off Travis though, that seems to be a whole other ordeal)

**Pull request merged**